File: Survey2-Functional/same.clj
; Function: equalsimp
; Language: Clojure
; Author: Kenneth Redillas
; Team: Haskers


(defn same [list1, list2]
      (cond
      ((empty? list1) (empty? list2))
      ((empty? list2) false)
      ((= (first list1) (first list2))
      	  (same (rest list1) (rest list2)))
	  :else false)
	  )

File: Survey2-Functional/same.hs
-- Function: equalsimp
-- Language: Haskell
-- Author: Kenneth Redillas
-- Team: Haskers

same list1 list2 =
     if (length list1 == 0) && (length list2 == 0) then True
     else if length list2 == 0 then False
     else if (head list1) == (head list2)
     	  then same (tail list1) (tail list2)
     else False

main = print (same ([1, 2, 3]) ([1, 2, 3]))
	